<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class> <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1400</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>NeuRPi - Experiment Control Interface</string>
  </property>
  <!-- <property name="windowFlags">
   <set>Qt::FramelessWindowHint</set>
  </property> -->
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="centralwidget_layout">
    <item>
     <widget class="QTabWidget" name="tabs">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="control_panel">
       <attribute name="title">
        <string>Control Panel</string>
       </attribute>
       <layout class="QVBoxLayout" name="control_panel_layout">
        <item>
         <widget class="QSplitter" name="main_splitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>

          <!-- Left Panel: Experiment Setup -->
          <widget class="QWidget" name="experiment_panel">
           <layout class="QVBoxLayout" name="experiment_panel_layout">

            <!-- Protocol Setup Group -->
            <item>
             <widget class="QGroupBox" name="protocol_setup_group">
              <property name="title">
               <string>Protocol Setup</string>
              </property>              <layout class="QGridLayout" name="protocol_setup_layout">
               <item row="0" column="0">
                <widget class="QLabel" name="rig_id_label">
                 <property name="text">
                  <string>Rig ID:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="rig_id">
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <item>
                  <property name="text">
                   <string>SELECT</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="protocol_label">
                 <property name="text">
                  <string>Protocol:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="protocol">
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <item>
                  <property name="text">
                   <string>SELECT</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="experiment_label">
                 <property name="text">
                  <string>Experiment:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QComboBox" name="experiment">
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <item>
                  <property name="text">
                   <string>SELECT</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="configuration_label">
                 <property name="text">
                  <string>Configuration:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="configuration">
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <item>
                  <property name="text">
                   <string>SELECT</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="response_mode_label">
                 <property name="text">
                  <string>Response Mode:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QComboBox" name="response_mode">
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <item>
                  <property name="text">
                   <string>SELECT</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Subject Setup Group -->
            <item>
             <widget class="QGroupBox" name="subject_setup_group">
              <property name="title">
               <string>Subject Setup</string>
              </property>
              <layout class="QGridLayout" name="subject_setup_layout">
               <item row="0" column="0">
                <widget class="QLabel" name="subject_name_label">
                 <property name="text">
                  <string>Subject Name:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QTextEdit" name="subject_name">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>40</height>
                  </size>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <property name="inputMethodHints">
                  <set>Qt::ImhUppercaseOnly</set>
                 </property>
                 <property name="tabChangesFocus">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QPushButton" name="create_new_subject">
                 <property name="text">
                  <string>New Subject</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>10</pointsize>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="subject_weight_label">
                 <property name="text">
                  <string>Subject Weight:</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>12</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QTextEdit" name="subject_weight">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>40</height>
                  </size>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>11</pointsize>
                  </font>
                 </property>
                 <property name="tabChangesFocus">
                  <bool>true</bool>
                 </property>
                 <property name="placeholderText">
                  <string>00.0</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Session Control Group -->
            <item>
             <widget class="QGroupBox" name="session_control_group">
              <property name="title">
               <string>Session Control</string>
              </property>
              <layout class="QHBoxLayout" name="session_control_layout">
               <item>
                <widget class="QPushButton" name="start_experiment">
                 <property name="text">
                  <string>Start Experiment</string>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>14</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>150</width>
                   <height>50</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string>QPushButton {
    background-color: #4CAF50;
    color: white;
    border: none;
    border-radius: 5px;
    padding: 10px;
}
QPushButton:hover {
    background-color: #45a049;
}
QPushButton:pressed {
    background-color: #3d8b40;
}</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Spacer to push groups to top -->
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>

           </layout>
          </widget>

          <!-- Right Panel: Status and Information -->
          <widget class="QWidget" name="status_panel">
           <layout class="QVBoxLayout" name="status_panel_layout">

            <!-- Session Information Group -->
            <item>
             <widget class="QGroupBox" name="session_info_group">
              <property name="title">
               <string>Session Information</string>
              </property>
              <layout class="QVBoxLayout" name="session_info_layout">
               <item>
                <widget class="QTextEdit" name="session_info_display">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>200</height>
                  </size>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                 <property name="font">
                  <font>
                   <family>Consolas</family>
                   <pointsize>10</pointsize>
                  </font>
                 </property>
                 <property name="placeholderText">
                  <string>Session information will appear here...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Active Rigs Group -->
            <item>
             <widget class="QGroupBox" name="active_rigs_group">
              <property name="title">
               <string>Active Rigs</string>
              </property>
              <layout class="QVBoxLayout" name="active_rigs_layout">
               <item>
                <widget class="QListWidget" name="active_rigs_list">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>150</height>
                  </size>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Log Output Group -->
            <item>
             <widget class="QGroupBox" name="log_group">
              <property name="title">
               <string>System Log</string>
              </property>
              <layout class="QVBoxLayout" name="log_layout">
               <item>
                <widget class="QTextEdit" name="log_display">
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                 <property name="font">
                  <font>
                   <family>Consolas</family>
                   <pointsize>9</pointsize>
                  </font>
                 </property>
                 <property name="placeholderText">
                  <string>System messages and logs will appear here...</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="log_controls_layout">
                 <item>
                  <spacer name="log_spacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QPushButton" name="clear_log_btn">
                   <property name="text">
                    <string>Clear Log</string>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>100</width>
                     <height>30</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>           </layout>
          </widget>

         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>protocol</tabstop>
  <tabstop>experiment</tabstop>
  <tabstop>configuration</tabstop>
  <tabstop>rig_id</tabstop>
  <tabstop>response_mode</tabstop>
  <tabstop>subject_name</tabstop>
  <tabstop>create_new_subject</tabstop>
  <tabstop>subject_weight</tabstop>
  <tabstop>start_experiment</tabstop>
  <tabstop>clear_log_btn</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
